src/dsp_tools/analyse_xml_data/extract_links_from_XML.py

```diff
@@ -23,85 +25,145 @@ def create_classes_from_root(root: etree._Element) -> list[ResptrLink] and list[
 
 def _create_classes_single_resource(
     resource: etree._Element,
-) -> list[ResptrLink] | None and list[XMLLink] | None and list[str] | None:
+) -> tuple[list[ResptrLink] | None, list[XMLLink] | None, list[str] | None]:
     subject_id = resource.attrib.get("id")
+    if not subject_id:
```

Comment:
this feels redundant, because we assume that a valid XML according to our xsd is uploaded. There the attribute "id" is mandatory. Instead of checking it everytime, I would  build a validation according to our xsd into the code.

Reply:
I don't know the exact context here, but normally, we don't only **assume** that the XML is valid - we **know** that it has been validated against the XSD.